pkg/api/handlers/compat/images_build.go

```diff
@@ -167,25 +168,54 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		addCaps = m
 	}
 
-	// convert addcaps formats
-	containerFiles := []string{}
-	if _, found := r.URL.Query()["dockerfile"]; found {
-		var m = []string{}
-		if err := json.Unmarshal([]byte(query.Dockerfile), &m); err != nil {
-			// it's not json, assume just a string
-			m = []string{filepath.Join(contextDirectory, query.Dockerfile)}
+	// If query paramemter `dockerfile` is set containerFiles variable will get overrided.
```

Comment:
This comment doesn't describe what's actually happening. I see no checks for a query parameter in this block.